pkg/grpc/grpc.go

```diff
@@ -0,0 +1,238 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	base "cosmossdk.io/api/cosmos/base/tendermint/v1beta1"
+
+	"github.com/archway-network/validator-exporter/pkg/config"
+	"github.com/archway-network/validator-exporter/pkg/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/bech32"
+	"github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	slashing "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	staking "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"google.golang.org/grpc"
+
+	log "github.com/archway-network/validator-exporter/pkg/logger"
+)
+
+const valConsStr = "valcons"
+
+type Client struct {
+	ctx       context.Context
+	ctxCancel context.CancelFunc
+	cfg       config.Config
+	conn      *grpc.ClientConn
+	connClose func()
+}
+
+func NewClient(cfg config.Config) (Client, error) {
+	client := Client{
+		cfg: cfg,
+	}
+
+	ctx, cancel := context.WithTimeout(
+		context.Background(),
+		time.Duration(client.cfg.Timeout)*time.Second,
+	)
+
+	client.ctx = ctx
+	client.ctxCancel = cancel
+
+	conn, err := cfg.GRPCConn()
+	if err != nil {
+		return Client{}, err
+	}
+
+	client.conn = conn
+	client.connClose = func() {
+		if err := conn.Close(); err != nil {
+			log.Error(fmt.Sprintf("failed to close connection :%s", err))
+		}
+	}
+
+	return client, nil
+}
+
+func (c Client) SignigInfos() ([]slashing.ValidatorSigningInfo, error) {
+	infos := []slashing.ValidatorSigningInfo{}
+	key := []byte{}
+	client := slashing.NewQueryClient(c.conn)
+
+	for {
+		request := &slashing.QuerySigningInfosRequest{Pagination: &query.PageRequest{Key: key}}
+
+		slashRes, err := client.SigningInfos(c.ctx, request)
+		if err != nil {
+			return nil, err
+		}
+
+		if slashRes == nil {
+			return nil, fmt.Errorf("got empty response from signing infos endpoint")
```

Comment:
I think you should use static wrapped errors and not dynamic ones in pkg/libraries.
It makes the logging/debugging a lot easier, example:
https://github.com/archway-network/discord-alertmanager/blob/main/pkg/discordbot/discordbot.go#L122-L126

This applies to all the packages that are returning errors